pkg/sql/catalog/descs/collection.go

```diff
@@ -40,21 +41,18 @@ import (
 func makeCollection(
 	leaseMgr *lease.Manager,
 	settings *cluster.Settings,
+	codec keys.SQLCodec,
 	hydratedTables *hydratedtables.Cache,
 	virtualSchemas catalog.VirtualSchemas,
 	temporarySchemaProvider TemporarySchemaProvider,
 ) Collection {
-	codec := keys.SystemSQLCodec
-	if leaseMgr != nil { // permitted for testing
-		codec = leaseMgr.Codec()
-	}
```

Comment:
I changed this because we now want to be able to create a Collection even when we don't have a lease manager. In particular, when we just have a tree.EvalContext and we just have the codec and the settings but we don't want to rely on the EvalPlanner (which might or might not have a useful implementation).